tests/test_temporal_dynamic.py

```diff
@@ -0,0 +1,96 @@
+from datetime import timedelta
+
+import pytest
+from temporalio import workflow
+from temporalio.testing import WorkflowEnvironment
+from temporalio.worker import Worker
+from temporalio.workflow import ActivityConfig
+
+from pydantic_ai import Agent, FunctionToolset
+from pydantic_ai.durable_exec.temporal import PydanticAIPlugin, TemporalAgent, TemporalFunctionToolset
+from pydantic_ai.models.test import TestModel
+
+
+# 1. Define Tool
+def echo(x: str) -> str:
+    return f'echo: {x}'
+
+
+# 2. Create Toolset with specific ID
+# Using an explicit ID allows us to reference it later if needed,
+# though here we pass the toolset object directly.
+toolset = FunctionToolset(tools=[echo], id='my_tools')
+
+# 3. Wrap Toolset for Temporal (DouweM pattern)
+wrapped_toolset = TemporalFunctionToolset(
+    toolset,
+    activity_name_prefix='shared_tools',
+    activity_config=ActivityConfig(start_to_close_timeout=timedelta(minutes=1)),
+    tool_activity_config={},
+    deps_type=type(None),
+)
+
+# 4. Create base agent for model activity registration
+# This agent's activities will be registered in the Worker
+# We use a known name "test_agent" so the dynamic agent can share it.
+base_model = TestModel()
+base_agent = Agent(base_model, name='test_agent')
+base_temporal_agent = TemporalAgent(base_agent)
```

Comment:
I want to also support the same kind of pre-registering we're adding in https://github.com/pydantic/pydantic-ai/pull/3537 for models, so you'd pass `toolsets={'funcs': toolset, 'other': ...}` and then you can pass `run(toolsets=['funcs'])` or `run(toolsets=[toolset])` later, in addition to the instance you've pre-wrapped yourself.